hydra-access-controls/app/models/hydra/access_controls/permission.rb

```diff
@@ -58,9 +58,9 @@ def build_agent(name, type)
       raise "Can't build agent #{inspect}" unless name && type
       self.agent = case type
                    when 'group'
-                     Agent.new(::RDF::URI.new("#{GROUP_AGENT_URL_PREFIX}##{name}"))
+                     Agent.new(::RDF::URI.new("#{GROUP_AGENT_URL_PREFIX}##{URI.encode(name)}"))
```

Comment:
How about making this a method with `prefix` as an argument. You can use it on line 63 too.